crates/in_stream/src/trait.rs

```diff
@@ -0,0 +1,143 @@
+use crate::*;
+use std::{
+    fmt::Debug,
+    io::{Error, ErrorKind, Result},
+};
+use url2::prelude::*;
+
+/// dynamic type for passing around in_stream configuration
+/// this is a stopgap until rust allows better constraints on associated types
+pub type InStreamConfigAny = Box<dyn std::any::Any + 'static + Send>;
```

Comment:
Would like a deeper explanation of this issue, can be done outside of the PR review however.